tests/crypto_policy_test.go

```diff
@@ -0,0 +1,240 @@
+	Expect(err).ToNot(HaveOccurred())
+	defer conn.Close()
+
+	certPool := x509.NewCertPool()
+	pemCert := getPemCertificate()
+	certPool.AppendCertsFromPEM(pemCert)
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			DialContext: func(ctx context.Context, network, addr string) (net.Conn, error) {
+				return conn, nil
+			},
+			TLSClientConfig: &tls.Config{CipherSuites: tlsConfig.CipherIDs(), MaxVersion: tlsConfig.MaxTLSVersion, RootCAs: certPool},
+		},
+	}
+
+	if subpath != "" {
+		subpath = "/" + subpath
+	}
+	attemptedUrl = fmt.Sprintf("https://ssp-webhook-service.kubevirt.svc:%s%s", port, subpath)
+	_, err = client.Get(attemptedUrl)
+	return attemptedUrl, err
+}
+
+func (c tlsConfigTestPermutation) testTLSEndpointAccessible(pod core.Pod, port, subpath string, tlsConfig clientTLSOptions) {
+	_, err := tryToAccessEndpoint(pod, port, subpath, tlsConfig)
+	Expect(err).ToNot(HaveOccurred(), "Can't access pod %s, at port %s, with tlsConfig %#v", pod.Name, port, tlsConfig)
+}
+
+func (c tlsConfigTestPermutation) testTLSEndpointNotAccessible(pod core.Pod, port, subpath string, tlsConfig clientTLSOptions) {
+	attemptedUrl, err := tryToAccessEndpoint(pod, port, subpath, tlsConfig)
+	expectedErrString1 := fmt.Sprintf("Get \"%s\": remote error: tls: protocol version not supported", attemptedUrl)
+	expectedErrString2 := fmt.Sprintf("Get \"%s\": remote error: tls: handshake failure", attemptedUrl)
+	Expect(err).To(SatisfyAny(MatchError(expectedErrString1), MatchError(expectedErrString2)), "Should not have been able to access pod %s, at port %s, with tlsConfig %#v, %#v, %#v", pod.Name, port, tlsConfig, tlsConfig.MaxTLSVersion, tlsConfig.CipherIDs())
+}
+
+func (c tlsConfigTestPermutation) testEndpointAccessabilityWithTLS(pod core.Pod, port, subpath string) {
+	for _, config := range c.allowedConfigs {
+		c.testTLSEndpointAccessible(pod, port, subpath, config)
+	}
+	for _, config := range c.disallowedConfigs {
+		c.testTLSEndpointNotAccessible(pod, port, subpath, config)
+	}
+}
+
+func testMetricsEndpoint(pod core.Pod, tlsConfig tlsConfigTestPermutation) {
+	tlsConfig.testEndpointAccessabilityWithTLS(pod, "8443", "metrics")
+}
+
+func testWebhookEndpoint(pod core.Pod, tlsConfig tlsConfigTestPermutation) {
+	tlsConfig.testEndpointAccessabilityWithTLS(pod, "9443", "")
+}
+
+func applyTLSConfig(tlsSecurityProfile *ocpv1.TLSSecurityProfile) {
+	watch, err := StartWatch(sspListerWatcher)
+	Expect(err).ToNot(HaveOccurred())
+	defer watch.Stop()
+
+	updateSsp(func(foundSsp *sspv1beta1.SSP) {
+		foundSsp.Spec.TLSSecurityProfile = tlsSecurityProfile
+	})
+	err = WatchChangesUntil(watch, isStatusDeploying, env.Timeout())
+	Expect(err).ToNot(HaveOccurred(), "SSP status should be deploying.")
+	err = WatchChangesUntil(watch, isStatusDeployed, env.Timeout())
+	Expect(err).ToNot(HaveOccurred(), "SSP status should be deployed.")
+}
+
+func testSettingTLSConfig(tlsConfigTestPermutation tlsConfigTestPermutation) {
```

Comment:
Can you inline this function? It is only used in one place and it is short.

Reply:
ok